tests/checkpatch_wrapper_test.sh

```diff
@@ -42,31 +59,73 @@ function oneTimeTearDown
   rm -rf "$FAKE_KERNEL"
 }
 
-function checkpatch
-{
-  res=$(execute_checkpatch "$FAKE_KERNEL/samples/codestyle_$1.c" 2>&1)
-  [[ "$res" == *"${!MSG[$1]}" ]]
-  assertTrue "Checkpatch should output: ${!MSG[$1]}" $?
-}
-
-function testWarning
+function warning_Test
 {
   checkpatch "warning"
 }
 
-function testError
+function error_Test
 {
   checkpatch "error"
 }
 
-function testChecks
+function checks_Test
 {
   checkpatch "check"
 }
 
-function testCorrect
+function correct_Test
 {
   checkpatch "correct"
 }
 
+function invalid_path_Test
+{
+  local build_fake_path
+  local output
+  local ret
+
+  build_fake_path="$RANDOM/$RANDOM/$RANDOM/xptolala"
+
+  output=$(execute_checkpatch "$build_fake_path")
+  ret="$?"
+
+  [[ "$ret" != 22 ]] && fail "We forced an invalid path and we expect an error"
+}
+
+function no_kernel_directory_Test
+{
```

Comment:
I really like this test! What do you think of adding one more to test `kw c` execution when the working dir is a kernel dir, but the given path is not? I.e. suppose I run `kw c /tmp/0001-testing-commit.patch` inside a kernel directory. Kw should succeed, using the `checkpatch.pl` script from the said dir. It would be nice to test this behavior, as well.

(We already test the case where the working dir is not a kernel dir but the given path is)

Reply:
I think that I did not get your idea. We already test a valid case indirectly in the first set of tests (error_Test, correct_Test, etc). Maybe I got the wrong idea? Are you talking about the path in front of `kw c`?

Reply:
In the fist set of tests we run `kw c` on a file _inside_ the kernel tree but _invoking from outside_ . It would be nice to have another test for the other way arround: i.e. running `kw c` _from within_ a kernel tree and passing a file that is _outside_. Something like:

```shell
# pseudocode:
local fake_kernel_tree = tests/.tmp
local not_a_kernel_tree = tests/.tmp2
mk_fake_kernel_root "$fake_kernel_tree"
cp -r tests/samples "$not_a_kernel_tree"

cd "$fake_kernel_tree"
$(execute_checkpatch) "$not_a_kernel_tree"
```